src/core/lib/layer-manager.js

```diff
@@ -366,26 +388,28 @@ export default class LayerManager {
   }
 
   // Rebuilds viewports from descriptors towards a certain window size
-  _rebuildViewportsFromViews({views, width, height}) {
-    const newViewports = views.map(view => view.makeViewport({width, height}));
+  _rebuildViewportsFromViews() {
+    if (this.viewsChanged) {
+      const {width, height, views, viewState} = this;
+      const newViewports = views.map(view => view.makeViewport({width, height, viewState}));
 
-    this.setNeedsRedraw('Viewport(s) changed');
+      this.setNeedsRedraw('Viewport(s) changed');
 
-    // Ensure one viewport is activated, layers may expect it
-    // TODO - handle empty viewport list (using dummy viewport), or assert
-    // const oldViewports = this.context.viewports;
-    // if (viewportsChanged) {
+      // Ensure one viewport is activated, layers may expect it
+      // TODO - handle empty viewport list (using dummy viewport), or assert
+      // const oldViewports = this.context.viewports;
 
-    const viewport = newViewports[0];
-    assert(viewport instanceof Viewport, 'Invalid viewport');
+      const viewport = newViewports[0];
+      assert(viewport instanceof Viewport, 'Invalid viewport');
 
-    this.context.viewports = newViewports;
-    this._activateViewport(viewport);
-    // }
+      this.context.viewports = newViewports;
+      this._activateViewport(viewport);
```

Comment:
Doesn't this trigger layer updates? Then called again in `drawLayers`?

Reply:
Yes, this is an area that has needed improvement since we introduced multi-viewport.

The reason it is here is that there is an issue when no viewport is selected in the layers during an update. 

I don't think it is a regression. Updated #1370 to make sure we address after this lands.